contracts/leaser/src/migrate.rs

```diff
@@ -2,101 +2,305 @@ use lease::api::MigrateMsg;
 use platform::batch::Batch;
 use sdk::cosmwasm_std::Addr;
 
-use crate::{error::ContractError, result::ContractResult};
+use crate::{msg::NbInstances, result::ContractResult};
 
-pub fn migrate_leases<I>(leases: I, lease_code_id: u64) -> ContractResult<Batch>
+pub struct Customer<LeaseIter> {
+    customer: Addr,
+    leases: LeaseIter,
+}
+
+#[derive(Default)]
+#[cfg_attr(test, derive(Debug, Eq, PartialEq))]
```

Comment:
I'd suggest dropping the conditional compilation for those. Either way they will be removed either by the compiler or linker.
```suggestion
#[derive(Debug, Eq, PartialEq, Default)]
```

Reply:
the point here is that they slow down the compilation ... and better describe what is allowed for the types and from there

Reply:
and that is the policy throughout the whole project ... let's not change it just for a type. If we decide later to do it we are going to prepare a dedicated commit/PR and compare the compilation time and binary size